File: AdvantageFramework.Core/Database/Entities/Alert.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("ALERT")]
    public partial class Alert
    {
        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        public Alert()
        {
            AlertAttachments = new HashSet<AlertAttachment>();
            AlertComments = new HashSet<AlertComment>();
            AlertEmpHours = new HashSet<AlertEmployeeHour>();
            AlertRcptDismisseds = new HashSet<AlertRecipientDismissed>();
            AlertRcptExts = new HashSet<AlertRecipientExternal>();
            AlertRcpts = new HashSet<AlertRecipient>();
        }

        [Key]
        [Column("ALERT_ID")]
        public int AlertId { get; set; }
        [Column("ALERT_TYPE_ID")]
        public int AlertTypeId { get; set; }
        [Column("ALERT_CAT_ID")]
        public int AlertCatId { get; set; }
        [Column("SUBJECT")]
        [StringLength(254)]
        public string Subject { get; set; }
        [Column("BODY", TypeName = "text")]
        public string Body { get; set; }
        [Column("GENERATED", TypeName = "smalldatetime")]
        public DateTime? Generated { get; set; }
        [Column("PRIORITY")]
        public short? Priority { get; set; }
        [Column("CL_CODE")]
        [StringLength(6)]
        public string ClCode { get; set; }
        [Column("DIV_CODE")]
        [StringLength(6)]
        public string DivCode { get; set; }
        [Column("PRD_CODE")]
        [StringLength(6)]
        public string PrdCode { get; set; }
        [Column("CMP_CODE")]
        [StringLength(6)]
        public string CmpCode { get; set; }
        [Column("JOB_NUMBER")]
        public int? JobNumber { get; set; }
        [Column("JOB_COMPONENT_NBR")]
        public short? JobComponentNbr { get; set; }
        [Column("ESTIMATE_NUMBER")]
        public int? EstimateNumber { get; set; }
        [Column("EST_COMPONENT_NBR")]
        public short? EstComponentNbr { get; set; }
        [Column("EST_QUOTE_NBR")]
        public short? EstQuoteNbr { get; set; }
        [Column("ESTIMATE_REV_NBR")]
        public short? EstimateRevNbr { get; set; }
        [Column("VN_CODE")]
        [StringLength(6)]
        public string VnCode { get; set; }
        [Column("EMP_CODE")]
        [StringLength(6)]
        public string EmpCode { get; set; }
        [Column("PO_NUMBER")]
        public int? PoNumber { get; set; }
        [Column("PO_REVISION")]
        public short? PoRevision { get; set; }
        [Column("ORDER_NBR")]
        public int? OrderNbr { get; set; }
        [Column("REV_NBR")]
        public short? RevNbr { get; set; }
        [Column("ALERT_USER")]
        [StringLength(100)]
        public string AlertUser { get; set; }
        [Column("TEMP_PDF_PATH")]
        [StringLength(254)]
        public string TempPdfPath { get; set; }
        [Column("ALERT_LEVEL")]
        [StringLength(50)]
        public string AlertLevel { get; set; }
        [Column("CMP_IDENTIFIER")]
        public int? CmpIdentifier { get; set; }
        [Column("OFFICE_CODE")]
        [StringLength(4)]
        public string OfficeCode { get; set; }
        [Column("BODY_HTML", TypeName = "text")]
        public string BodyHtml { get; set; }
        [Column("CP_ALERT")]
        public short? CpAlert { get; set; }
        [Column("BA_BATCH_ID")]
        public int? BaBatchId { get; set; }
        [Column("TASK_SEQ_NBR")]
        public short? TaskSeqNbr { get; set; }
        [Column("DUE_DATE", TypeName = "smalldatetime")]
        public DateTime? DueDate { get; set; }
        [Column("ALERT_STATE_ID")]
        public int? AlertStateId { get; set; }
        [Column("ALRT_NOTIFY_HDR_ID")]
        public int? AlrtNotifyHdrId { get; set; }
        [Column("TIME_DUE")]
        [StringLength(10)]
        public string TimeDue { get; set; }
        [Column("VER")]
        [StringLength(10)]
        public string Ver { get; set; }
        [Column("BUILD")]
        [StringLength(10)]
        public string Build { get; set; }
        [Column("ALERT_SEQ_NBR")]
        public short? AlertSeqNbr { get; set; }
        [Column("SENT", TypeName = "smalldatetime")]
        public DateTime? Sent { get; set; }
        [Column("LAST_UPDATED", TypeName = "smalldatetime")]
        public DateTime? LastUpdated { get; set; }
        [Column("VER2")]
        [StringLength(10)]
        public string Ver2 { get; set; }
        [Column("BUILD2")]
        [StringLength(10)]
        public string Build2 { get; set; }
        [Column("ALERT_USER_CP")]
        public int? AlertUserCp { get; set; }
        [Column("NON_TASK_ID")]
        public int? NonTaskId { get; set; }
        [Column("AP_ID")]
        public int? ApId { get; set; }
        [Column("AP_SEQ")]
        public short? ApSeq { get; set; }
        [Column("ATB_REV_ID")]
        public int? AtbRevId { get; set; }
        [Column("IS_DRAFT")]
        public bool? IsDraft { get; set; }
        [Column("ATTACHMENT_COUNT")]
        public int? AttachmentCount { get; set; }
        [Column("LAST_UPDATED_USER_CODE")]
        [StringLength(100)]
        public string LastUpdatedUserCode { get; set; }
        [Column("LAST_UPDATED_FML")]
        [StringLength(80)]
        public string LastUpdatedFml { get; set; }
        [Column("ASSIGNED_EMP_CODE")]
        [StringLength(6)]
        public string AssignedEmpCode { get; set; }
        [Column("ASSIGNED_EMP_FML")]
        [StringLength(80)]
        public string AssignedEmpFml { get; set; }
        [Column("ASSIGN_COMPLETED")]
        public bool? AssignCompleted { get; set; }
        [Column("LAST_ASSIGNED_EMP_CODE")]
        [StringLength(6)]
        public string LastAssignedEmpCode { get; set; }
        [Column("JOB_VER_HDR_ID")]
        public int? JobVerHdrId { get; set; }
        [Column("IS_CS_REVIEW")]
        public bool? IsCsReview { get; set; }
        [Column("CS_AUTO_APPR_METHOD")]
        public int? CsAutoApprMethod { get; set; }
        [Column("CS_REVIEW_TYPE")]
        public int? CsReviewType { get; set; }
        [Column("CS_REVIEW_STATUS")]
        public int? CsReviewStatus { get; set; }
        [Column("CS_PROJECT_ID")]
        public int? CsProjectId { get; set; }
        [Column("CS_REVIEW_ID")]
        public int? CsReviewId { get; set; }
        [Column("CS_NUM_REVIEWER")]
        public int? CsNumReviewer { get; set; }
        [Column("CS_NUM_CMPLT")]
        public int? CsNumCmplt { get; set; }
        [Column("CS_ASSET_IMG", TypeName = "image")]
        public byte[] CsAssetImg { get; set; }
        [Column("CS_NUM_REJECT")]
        public int? CsNumReject { get; set; }
        [Column("CS_NUM_DEFER")]
        public int? CsNumDefer { get; set; }
        [Column("CS_NUM_APPR")]
        public int? CsNumAppr { get; set; }
        [Column("IS_WORK_ITEM")]
        public bool? IsWorkItem { get; set; }
        [Column("BOARD_STATE_ID")]
        public int? BoardStateId { get; set; }
        [Column("EPIC_ID")]
        public int? EpicId { get; set; }
        [Column("HRS_ALLOWED", TypeName = "decimal(7, 2)")]
        public decimal? HrsAllowed { get; set; }
        [Column("BACKLOG_SEQ_NBR")]
        public int? BacklogSeqNbr { get; set; }
        [Column("START_DATE", TypeName = "smalldatetime")]
        public DateTime? StartDate { get; set; }

        [ForeignKey(nameof(AlertCatId))]
        [InverseProperty(nameof(AlertCategory.Alerts))]
        public virtual AlertCategory AlertCat { get; set; }
        [ForeignKey(nameof(AlertStateId))]
        [InverseProperty("Alerts")]
        public virtual AlertState AlertState { get; set; }
        [ForeignKey(nameof(AlertTypeId))]
        [InverseProperty("Alerts")]
        public virtual AlertType AlertType { get; set; }
        [InverseProperty(nameof(AlertAttachment.Alert))]
        public virtual ICollection<AlertAttachment> AlertAttachments { get; set; }
        [InverseProperty(nameof(AlertComment.Alert))]
        public virtual ICollection<AlertComment> AlertComments { get; set; }
        [InverseProperty(nameof(AlertEmployeeHour.Alert))]
        public virtual ICollection<AlertEmployeeHour> AlertEmpHours { get; set; }
        [InverseProperty(nameof(AlertRecipientDismissed.Alert))]
        public virtual ICollection<AlertRecipientDismissed> AlertRcptDismisseds { get; set; }
        [InverseProperty(nameof(AlertRecipientExternal.Alert))]
        public virtual ICollection<AlertRecipientExternal> AlertRcptExts { get; set; }
        [InverseProperty(nameof(AlertRecipient.Alert))]
        public virtual ICollection<AlertRecipient> AlertRcpts { get; set; }
        [InverseProperty(nameof(ClientPortalAlertRecipient.Alert))]
        public virtual ICollection<ClientPortalAlertRecipient> ClientPortalAlertRecipients { get; set; }
        [InverseProperty(nameof(MediaRFPHeader.Alert))]
        public virtual ICollection<MediaRFPHeader> MediaRfpHeaders { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/Alert.vb
Namespace Database.Entities

	<Table("ALERT")>
	Public Class Alert
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

		Public Enum Properties
			ID
			AlertTypeID
			AlertCategoryID
			Subject
			Body
			GeneratedDate
			PriorityLevel
			EmployeeCode
			UserCode
			AlertLevel
			EmailBody
			AlertStateID
			AlertAssignmentTemplateID
			DueDate
			TimeDue
			ClientContactID
			ClientCode
			DivisionCode
			ProductCode
			CampaignCode
			JobNumber
			JobComponentNumber
			EstimateNumber
			EstimateComponentNumber
			EstimateQuoteNumber
			EstimateRevisionNumber
			VendorCode
			PORevisionNumber
			PONumber
			OfficeCode
			CampaignID
			NonTaskID
			IsCPAlert
			BillingApprovalBatchID
			TaskSequenceNumber
			AlertSequenceNumber
			Version
			Build
			Version2
			Build2
			AccountPayableID
			AccountPayableSequenceNumber
			MediaATBRevisionID
            LastUpdated
            JobVersionID
            IsDraft
            AttachmentCount
            LastUpdatedUserCode
            LastUpdatedFullName
            AssignedEmployeeCode
            AssignedEmployeeFullName
            AssignmentCompleted
            LastAssignedEmployeeCode
            IsConceptShareReview
            ConceptShareAutoApproveMethodID
            ConceptShareReviewTypeID
            ConceptShareReviewStatusID
            ConceptShareProjectID
            ConceptShareReviewID
            ConceptShareNumberOfReviewers
            ConceptShareNumberOfCompletedReviewers
            ConceptShareNumberOfApprovedReviewers
            ConceptShareNumberOfRejectedReviewers
            ConceptShareNumberOfDeferredReviewers
            ConceptShareAssetImage
            IsWorkItem
            BoardStateID
            EpicID
            HoursAllowed
            BacklogSequenceNumber
            StartDate

            AlertType
			AlertAttachments
			AlertComments
			AlertRecipients
			AlertCategory
			AlertRecipientDismisseds
			ClientPortalAlertRecipients
			ClientPortalAlertRecipientDismisseds
			AlertState

		End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.None)>
        <Required>
		<Column("ALERT_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ID() As Integer
		<Required>
		<Column("ALERT_TYPE_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertTypeID() As Integer
		<Required>
		<Column("ALERT_CAT_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertCategoryID() As Integer
		<MaxLength(254)>
		<Column("SUBJECT", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Subject() As String
		<Column("BODY")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Body() As String
		<Column("GENERATED")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property GeneratedDate() As Nullable(Of Date)
		<Column("PRIORITY")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property PriorityLevel() As Nullable(Of Short)
		<MaxLength(6)>
		<Column("EMP_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EmployeeCode() As String
		<MaxLength(100)>
		<Column("ALERT_USER", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property UserCode() As String
		<MaxLength(50)>
		<Column("ALERT_LEVEL", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertLevel() As String
		<Column("BODY_HTML")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EmailBody() As String
		<Column("ALERT_STATE_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertStateID() As Nullable(Of Integer)
		<Column("ALRT_NOTIFY_HDR_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertAssignmentTemplateID() As Nullable(Of Integer)
		<Column("DUE_DATE")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property DueDate() As Nullable(Of Date)
		<MaxLength(10)>
		<Column("TIME_DUE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property TimeDue() As String
		<Column("ALERT_USER_CP")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ClientContactID() As Nullable(Of Integer)
		<MaxLength(6)>
		<Column("CL_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ClientCode() As String
		<MaxLength(6)>
		<Column("DIV_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property DivisionCode() As String
		<MaxLength(6)>
		<Column("PRD_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ProductCode() As String
		<MaxLength(6)>
		<Column("CMP_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property CampaignCode() As String
		<Column("JOB_NUMBER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property JobNumber() As Nullable(Of Integer)
		<Column("JOB_COMPONENT_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property JobComponentNumber() As Nullable(Of Short)
		<Column("ESTIMATE_NUMBER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EstimateNumber() As Nullable(Of Integer)
		<Column("EST_COMPONENT_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EstimateComponentNumber() As Nullable(Of Short)
		<Column("EST_QUOTE_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EstimateQuoteNumber() As Nullable(Of Short)
		<Column("ESTIMATE_REV_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property EstimateRevisionNumber() As Nullable(Of Short)
		<MaxLength(6)>
		<Column("VN_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property VendorCode() As String
		<Column("PO_REVISION")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property PORevisionNumber() As Nullable(Of Short)
		<Column("PO_NUMBER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property PONumber() As Nullable(Of Integer)
		<MaxLength(4)>
		<Column("OFFICE_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property OfficeCode() As String
		<Column("CMP_IDENTIFIER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property CampaignID() As Nullable(Of Integer)
		<Column("NON_TASK_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property NonTaskID() As Nullable(Of Integer)
		<Column("CP_ALERT")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property IsCPAlert() As Nullable(Of Short)
		<Column("BA_BATCH_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property BillingApprovalBatchID() As Nullable(Of Integer)
		<Column("TASK_SEQ_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property TaskSequenceNumber() As Nullable(Of Short)
		<Column("ALERT_SEQ_NBR")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AlertSequenceNumber() As Nullable(Of Short)
		<MaxLength(10)>
		<Column("VER", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Version() As String
		<MaxLength(10)>
		<Column("BUILD", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Build() As String
		<MaxLength(10)>
		<Column("VER2", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Version2() As String
		<MaxLength(10)>
		<Column("BUILD2", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Build2() As String
		<Column("AP_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AccountPayableID() As Nullable(Of Integer)
		<Column("AP_SEQ")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property AccountPayableSequenceNumber() As Nullable(Of Short)
		<Column("ATB_REV_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property MediaATBRevisionID() As Nullable(Of Integer)
		<Column("LAST_UPDATED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LastUpdated() As Nullable(Of Date)
        <Column("JOB_VER_HDR_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobVersionID() As Nullable(Of Integer)
        <Column("IS_DRAFT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsDraft() As Nullable(Of Boolean)
        <Column("ATTACHMENT_COUNT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AttachmentCount() As Nullable(Of Integer)
        <Column("LAST_UPDATED_USER_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LastUpdatedUserCode() As String
        <Column("LAST_UPDATED_FML", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LastUpdatedFullName() As String
        <Column("ASSIGNED_EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AssignedEmployeeEmployeeCode() As String
        <Column("ASSIGNED_EMP_FML", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AssignedEmployeeFullName() As String
        <Column("ASSIGN_COMPLETED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AssignmentCompleted() As Nullable(Of Boolean)
        <Column("LAST_ASSIGNED_EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LastAssignedEmployeeCode() As String
        <Column("IS_CS_REVIEW")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsConceptShareReview() As Nullable(Of Boolean)
        <Column("CS_AUTO_APPR_METHOD")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareAutoApproveMethodID() As Nullable(Of Integer)
        <Column("CS_REVIEW_TYPE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareReviewTypeID() As Nullable(Of Integer)
        <Column("CS_REVIEW_STATUS")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareReviewStatusID() As Nullable(Of Integer)
        <Column("CS_PROJECT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareProjectID() As Nullable(Of Integer)
        <Column("CS_REVIEW_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareReviewID() As Nullable(Of Integer)
        <Column("CS_NUM_REVIEWER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareNumberOfReviewers() As Nullable(Of Integer)
        <Column("CS_NUM_CMPLT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareNumberOfCompletedReviewers() As Nullable(Of Integer)
        <Column("CS_ASSET_IMG")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareAssetImage() As Byte()
        <Column("CS_NUM_REJECT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareNumberOfRejectedReviewers() As Nullable(Of Integer)
        <Column("CS_NUM_DEFER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareNumberOfDeferredReviewers() As Nullable(Of Integer)
        <Column("CS_NUM_APPR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ConceptShareNumberOfApprovedReviewers() As Nullable(Of Integer)
        <Column("IS_WORK_ITEM")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsWorkItem() As Nullable(Of Boolean)
        <Column("BOARD_STATE_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property BoardStateID() As Nullable(Of Integer)
        <Column("EPIC_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property EpicID() As Nullable(Of Integer)
        <Column("HRS_ALLOWED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property HoursAllowed() As Nullable(Of Decimal)
        <Column("BACKLOG_SEQ_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property BacklogSequenceNumber() As Nullable(Of Integer)
        <Column("START_DATE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property StartDate() As Nullable(Of Date)

        Public Overridable Property AlertType As Database.Entities.AlertType
        Public Overridable Property AlertAttachments As ICollection(Of Database.Entities.AlertAttachment)
        Public Overridable Property AlertComments As ICollection(Of Database.Entities.AlertComment)
        Public Overridable Property AlertRecipients As ICollection(Of Database.Entities.AlertRecipient)
        Public Overridable Property AlertCategory As Database.Entities.AlertCategory
        Public Overridable Property AlertRecipientDismisseds As ICollection(Of Database.Entities.AlertRecipientDismissed)
        Public Overridable Property ClientPortalAlertRecipients As ICollection(Of Database.Entities.ClientPortalAlertRecipient)
        Public Overridable Property ClientPortalAlertRecipientDismisseds As ICollection(Of Database.Entities.ClientPortalAlertRecipientDismissed)
        Public Overridable Property AlertState As Database.Entities.AlertState

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID.ToString

        End Function

#End Region

	End Class

End Namespace
